src/impl_aarch64_freebsd.c

```diff
@@ -0,0 +1,365 @@
+    ID_AA64ISAR0_SET_FEATURE(sha3, "SHA3"),
+    ID_AA64ISAR0_SET_FEATURE(sm3, "SM3"),
+    ID_AA64ISAR0_SET_FEATURE(sm4, "SM4"),
+    ID_AA64ISAR0_SET_FEATURE(aes_pmull, "AES+PMULL"),
+    ID_AA64ISAR0_SET_FEATURE(sha2_sha512, "SHA2+SHA512"),
+    ID_AA64ISAR0_SET_FEATURE(flagm, "CondM-8.4"),
+    ID_AA64ISAR0_SET_FEATURE(flagm2, "CondM-8.5"),
+    ID_AA64ISAR0_SET_FEATURE(rng, "RNG"),
+    MRS_FEATURE_END,
+};
+
+// ID_AA64ISAR1_EL1
+// -----------------------------------------------------------------------------
+ID_AA64ISAR1_FEATURES(i8mm, AARCH64_I8MM)
+ID_AA64ISAR1_FEATURES(dgh, AARCH64_DGH)
+ID_AA64ISAR1_FEATURES(bf16, AARCH64_BF16)
+ID_AA64ISAR1_FEATURES(sb, AARCH64_SB)
+ID_AA64ISAR1_FEATURES(frint, AARCH64_FRINT)
+ID_AA64ISAR1_FEATURES(pacg, AARCH64_PACG)
+ID_AA64ISAR1_FEATURES(paca, AARCH64_PACA)
+ID_AA64ISAR1_FEATURES(lrcpc, AARCH64_LRCPC)
+ID_AA64ISAR1_FEATURES(ilrcpc, AARCH64_LRCPC, AARCH64_ILRCPC)
+ID_AA64ISAR1_FEATURES(fcma, AARCH64_FCMA)
+ID_AA64ISAR1_FEATURES(jscvt, AARCH64_JSCVT)
+ID_AA64ISAR1_FEATURES(dcpop, AARCH64_DCPOP)
+ID_AA64ISAR1_FEATURES(dcpodp, AARCH64_DCPOP, AARCH64_DCPODP)
+
+static MrsFeature id_aa64isar1[] = {
+    ID_AA64ISAR1_SET_FEATURE(i8mm, "I8MM"),
+    ID_AA64ISAR1_SET_FEATURE(dgh, "DGH"),
+    ID_AA64ISAR1_SET_FEATURE(bf16, "BF16"),
+    ID_AA64ISAR1_SET_FEATURE(sb, "SB"),
+    ID_AA64ISAR1_SET_FEATURE(frint, "FRINTTS"),
+    ID_AA64ISAR1_SET_FEATURE(pacg, "GPI"),
+    ID_AA64ISAR1_SET_FEATURE(paca, "GPA"),
+    ID_AA64ISAR1_SET_FEATURE(lrcpc, "RCPC-8.3"),
+    ID_AA64ISAR1_SET_FEATURE(ilrcpc, "RCPC-8.4"),
+    ID_AA64ISAR1_SET_FEATURE(fcma, "FCMA"),
+    ID_AA64ISAR1_SET_FEATURE(jscvt, "JSCVT"),
+    ID_AA64ISAR1_SET_FEATURE(dcpop, "DCPoP"),
+    ID_AA64ISAR1_SET_FEATURE(dcpodp, "DCCVADP"),
+    MRS_FEATURE_END,
+};
+
+// ID_AA64ISAR2_EL1
+// -----------------------------------------------------------------------------
+ID_AA64ISAR2_FEATURES(rpres, AARCH64_RPRES)
+
+static MrsFeature id_aa64isar2[] = {
+    ID_AA64ISAR2_SET_FEATURE(rpres, "RPRES"),
+    MRS_FEATURE_END,
+};
+
+// ID_AA64PFR0_EL1
+// -----------------------------------------------------------------------------
+ID_AA64PFR0_FEATURES(fp, AARCH64_FP)
+ID_AA64PFR0_FEATURES(fphp, AARCH64_FP, AARCH64_FPHP)
+ID_AA64PFR0_FEATURES(asimd, AARCH64_ASIMD)
+ID_AA64PFR0_FEATURES(asimdhp, AARCH64_ASIMD, AARCH64_ASIMDHP)
+ID_AA64PFR0_FEATURES(dit, AARCH64_DIT)
+ID_AA64PFR0_FEATURES(sve, AARCH64_SVE)
+
+static MrsFeature id_aa64pfr0[] = {
+    ID_AA64PFR0_SET_FEATURE(fp, "FP"),
+    ID_AA64PFR0_SET_FEATURE(fphp, "FP+HP"),
+    ID_AA64PFR0_SET_FEATURE(asimd, "AdvSIMD"),
+    ID_AA64PFR0_SET_FEATURE(asimdhp, "AdvSIMD+HP"),
+    ID_AA64PFR0_SET_FEATURE(dit, "PSTATE.DIT"),
+    ID_AA64PFR0_SET_FEATURE(sve, "SVE"),
+    MRS_FEATURE_END,
+};
+
+// ID_AA64PFR1_EL1
+// -----------------------------------------------------------------------------
+ID_AA64PFR1_FEATURES(ssbs, AARCH64_SSBS)
+// TODO: Add ssbs2 detection
+ID_AA64PFR1_FEATURES(ssbs2, AARCH64_SSBS)
```

Comment:
SBS | Meaning
-- | --
0b0000 | AArch64 provides no mechanism to control the use of Speculative Store Bypassing.
0b0001 | AArch64 provides the PSTATE.SSBS mechanism to mark regions that are Speculative Store Bypass Safe.
0b0010 | AArch64 provides the PSTATE.SSBS mechanism to mark regions that are Speculative Store Bypassing Safe, and the MSR and MRS instructions to directly read and write the PSTATE.SSBS field.

`FEAT_SSBS` implements the functionality identified by the value 0b0001.
`FEAT_SSBS2` implements the functionality identified by the value 0b0010.

in current impl linux identifies ssbs2 as ssbs:

HWCAP_SSBS
Functionality implied by ID_AA64PFR1_EL1.SSBS == 0b0010.

refs: 
* https://developer.arm.com/documentation/ddi0601/2020-12/AArch64-Registers/ID-AA64PFR1-EL1--AArch64-Processor-Feature-Register-1
* https://docs.kernel.org/arm64/elf_hwcaps.html

so we should , add `ssbs2` new field, set to true `ssbs` for linux when ID_AA64PFR1_EL1.SSBS == 0b0010 is true, for freebsd just add here new `AARCH64_SSBS2`